primitives/npos-elections/src/tests.rs

```diff
@@ -18,12 +18,57 @@
 //! Tests for npos-elections.
 
 use crate::{
-	balancing, helpers::*, mock::*, seq_phragmen, seq_phragmen_core, setup_inputs, to_support_map,
-	Assignment, BalancingConfig, ElectionResult, ExtendedBalance, StakedAssignment, Support, Voter,
+	approval_voting::*, balancing, helpers::*, mock::*, seq_phragmen, seq_phragmen_core,
+	setup_inputs, to_support_map, Assignment, BalancingConfig, ElectionResult, ExtendedBalance,
+	StakedAssignment, Support, Voter,
 };
 use sp_arithmetic::{PerU16, Perbill, Percent, Permill};
 use substrate_test_utils::assert_eq_uvec;
 
+#[test]
+fn approval_voting_works() {
+	let candidates = vec![1, 2, 3, 4];
+	let voters = vec![(10, vec![1, 2]), (20, vec![1, 2]), (30, vec![1, 2, 3]), (40, vec![4])];
+	let stake_of = create_stake_of(&[(10, 10), (20, 20), (30, 30), (40, 40)]);
+
+	let voters = voters
+		.iter()
+		.map(|(ref v, ref vs)| (*v, stake_of(v), vs.clone()))
+		.collect::<Vec<_>>();
+
+	let ElectionResult::<_, Perbill> { winners, assignments } =
+		approval_voting(3, candidates, voters).unwrap();
+
+	assert_eq_uvec!(winners, vec![(1, 60), (2, 60), (4, 40)]);
+	assert_eq_uvec!(
```

Comment:
@kianenigma I want to bring one thing up to make sure we're on the same page. The approval voting is not normalising the vote weights, as you can see in the test vecs below. i.e. each assignment's vote distribution is 100%, regardless of the number of candidates backed. The reasoning behind this is that each vote should be backed with 100% weight of the voter's stake for all voted candidates.

The caller can still normalise the votes after calling `fn approval_voting`, as it is demonstrated by these tests in the [election-provider-support](https://github.com/paritytech/substrate/blob/f4a0f341b7eb7973f183504743f7f76ad219c002/frame/election-provider-support/src/onchain.rs#L353).

This is the behaviour I expect form an approval voting scheme and added a [comment about it](https://github.com/paritytech/substrate/blob/f4a0f341b7eb7973f183504743f7f76ad219c002/primitives/npos-elections/src/approval_voting.rs#L30).

Reply:
Given the fact that you have opted to give everyone 100%, it is fine either way. As you pointed out normalize is a noop here because everyone is at 100%.